packages/editor/src/components/post-publish-button/index.js

```diff
@@ -147,7 +147,7 @@ export class PostPublishButton extends Component {
 		};
 
 		const buttonProps = {
-			'aria-disabled': isButtonDisabled && ! hasNonPostEntityChanges,
+			'aria-disabled': isButtonDisabled,
```

Comment:
I wonder if there would be any reason to have this condition `isButtonDisabled || hasNonPostEntityChanges`.  I interpret this old condition here as a logic error aimed at disabling the button when `hasNonPostEntityChanges` as well.

Doing so may be redundant, as I dont think there is any way to access the publish panel while `hasNonPostEntityChanges` is still true... but it may be an extra safety?

Reply:
Hmm, don't we want the button to be enabled (rather than disabled) if there are any unsaved non-post entity changes (even if there are no changes to the post itself), no? (Too many double negations here 😅)

---

Aside, these are the so-called `buttonProps` (for the Save/Publish button that actually saves/publishes the post). However, the same component can be used in the guise of the "Prepublish _toggle_" (which opens the Prepublish panel, IIUC), and for which the props are set a couple of lines below this:

https://github.com/WordPress/gutenberg/blob/8ed827d91d078bffd602755a0449ade569dd1d22/packages/editor/src/components/post-publish-button/index.js#L157-L164

Here, I kept the `isButtonDisabled && ! hasNonPostEntityChanges`, as it seemed to make sense to me: If all the other props indicate the toggle should be disabled (since there are no changes to the post that need saving), _and_ there are no non-post entity changes, _then_ we disable the button.

Reply:
> Hmm, don't we want the button to be enabled (rather than disabled) if there are any unsaved non-post entity changes (even if there are no changes to the post itself), no? 

Err yes, I was confusing myself thinking this was a different button.